ruby-challenges.rb

```diff
@@ -9,19 +9,52 @@
 # Expected output: ['coffee', 'soda water']
 filter_letter_t = 't'
 # Expected output: ['tea', 'water', 'soda water']
+# Notes: Used the select because it returns a new array made up of elements in the original array that return a true value in the provided code block
+def filter_arr (arr, str)
+    arr.select { |arr| arr.include?(str)}
+end
+
+
 
 
 # -------------------2) Create a method that takes in a hash and returns one array with all the hash values at their own index and in alphabetical order. No nested arrays. Use the test variable provided.
 # HINT: Google 'ruby get rid of nested arrays'
 
-us_states = { northwest: ['Washington', 'Oregon', 'Idaho'], southwest: ['California', 'Arizona', 'Nevada'], notheast: ['Maine', 'New Hampshire', 'Rhode Island'] }
+us_states = { northwest: ['Washington', 'Oregon', 'Idaho'], southwest: ['California', 'Arizona', 'Nevada'], northeast: ['Maine', 'New Hampshire', 'Rhode Island'] }
 # Expected output: ['Arizona', 'California', 'Idaho', 'Maine', 'Nevada', 'New Hampshire', 'Oregon', 'Rhode Island', 'Washington'] 
+def single_arr hash
+    new_hash = hash[:northwest] + hash[:northeast] + hash[:southwest]
```

Comment:
To make this method dynamic, how can you refactor? Right now, it only works if the keys in the hash are specifically northwest, northeast, and southwest, but what if the key names are different?